File: backend/schemas/transfer.py
from pydantic import BaseModel, ConfigDict

from backend.schemas.outbound import PositiveInt


class TransferCreate(BaseModel):
    model_config = ConfigDict(extra="forbid")

    lot_id: PositiveInt
    from_location_id: PositiveInt
    to_location_id: PositiveInt
    qty: PositiveInt


class TransferResult(BaseModel):
    lot_id: int
    from_location_id: int
    to_location_id: int
    from_qty: int
    to_qty: int
    movement_id: int


class TransferRecord(BaseModel):
    movement_id: int
    lot_id: int
    lot_code: str
    product_name: str
    unit: str
    from_location_id: int
    from_location_code: str
    to_location_id: int
    to_location_code: str
    qty: int
    actor_name: str
    created_at: str

File: frontend/src/api/transfer.ts
import { apiRequest } from './client';

export interface TransferInput {
  lot_id: number; from_location_id: number; to_location_id: number; qty: number;
}
export interface TransferResult {
  lot_id: number; from_location_id: number; to_location_id: number;
  from_qty: number; to_qty: number; movement_id: number;
}
export interface TransferRecord extends TransferInput {
  movement_id: number; lot_code: string; product_name: string; unit: string;
  from_location_code: string; to_location_code: string; actor_name: string; created_at: string;
}

export function submitTransfer(payload: TransferInput): Promise<TransferResult> {
  return apiRequest('/api/outbound/transfers', {
    method: 'POST', body: JSON.stringify(payload), signal: AbortSignal.timeout(15000),
  });
}

export function getTransferRecords(): Promise<TransferRecord[]> {
  return apiRequest('/api/outbound/transfers', { signal: AbortSignal.timeout(10000) });
}
